lms/static/js/ccx/schedule.js

```diff
@@ -251,9 +269,53 @@ var edx = edx || {};
 	    }
 	},
 
+	valid_dates: function(start, due) {
+          var errorMessage;
+          // Start date is compulsory and due date is optional.
+          if (_.isEmpty(start) && !_.isEmpty(due)) {
+            errorMessage = gettext("Please enter valid start date and time.");
+          } else if (!_.isEmpty(start) && !_.isEmpty(due)) {
+            var dueParts = self.parse_date(due);
+            var startParts = self.parse_date(start);
+            var dueDate = new Date(
+              dueParts[0],
+              dueParts[1],
+              dueParts[2],
+              dueParts[3],
+              dueParts[4]
+            );
+            var startDate = new Date(
+              startParts[0],
+              startParts[1],
+              startParts[2],
+              startParts[3],
+              startParts[4]
+						);
+            if (dueDate < startDate) {
+              errorMessage = gettext("Due date cannot be before start date.");
+            }
+          }
+          return errorMessage;
+	},
+
+	parse_date: function(date) {
```

Comment:
Gah, on my last review I was too focused on checking this line-by-line, and I didn't step back to see the larger picture. For parsing and manipulating dates and times, you should use the [moment.js](http://momentjs.com/) library, which is already being used in the platform. It looks like the format you're parsing is *almost* [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601), but you're using a space separator instead of a `T` separator between the date and time. Fortunately, this is very common, so moment.js should still handle it automatically. You can parse the string simply by doing:
```js
return moment(date)  // returns a Moment object, not an Array
```
If you want to transform that Moment object to a native Date object, or an array of integers like what you're doing now, you can call the [`asDate()`](http://momentjs.com/docs/#/displaying/as-javascript-date/) or [`asArray()`](http://momentjs.com/docs/#/displaying/as-array/) methods.